app/src/lib/download.ts

```diff
@@ -0,0 +1,25 @@
+import {getAuthHeader} from "./auth/getAuthHeader";
+
+export const download = async (url: string, filename: string, authRequired = true) => {
+    const headers = authRequired ? getAuthHeader(): undefined;
```

Comment:
can use existing fetcher to achieve all this data fetching pretty sure....

Reply:
Yeah, I wondered about that, but would need to modify the existing fetcher, which currently just returns JSON. I figured it wouldn't actually share very much code other than appending the auth header (since this has to deal with a blob on success and json on failure). I could put in the capacity to the current fetcher to allow blob return but I think that would overcompilicate that probably..

Reply:
ohh didnt realize it does a blob on the res directly.... yup good